VK_App/VK_App/NetworkService/PhotoCacheService.swift

```diff
@@ -0,0 +1,127 @@
+                at: url,
+                withIntermediateDirectories: true,
+                attributes: nil
+            )
+        }
+        return pathName
+    }()
+
+    private var images = [String: UIImage]()
+
+    // MARK: - Initializers
+
+    init(container: UITableViewController) {
+        self.container = Table(table: container)
+    }
+
+    // MARK: - Public Properties
+
+    func photo(atIndexPath indexPath: IndexPath, byUrl url: String) -> UIImage? {
+        var image: UIImage?
+        if let photo = images[url] {
+            image = photo
+        } else if let photo = getImageFromCache(url: url) {
+            image = photo
+        } else {
+            loadPhoto(atIndexPath: indexPath, byUrl: url)
+        }
+        return image
+    }
+
+    // MARK: - Private Methods
+
+    private func getFilePath(url: String) -> String? {
+        guard let cachesDirectory = FileManager.default.urls(
+            for:
+            .cachesDirectory,
+            in: .userDomainMask
+        ).first else { return nil }
+        let hashName = url.split(separator: Constants.slashChar).last ?? Constants.defaultText
+        return cachesDirectory.appendingPathComponent(
+            PhotoCacheService.pathName +
+                Constants.slashString + hashName
+        ).path
+    }
+
+    private func saveImageToCache(url: String, image: UIImage) {
+        guard let fileName = getFilePath(url: url),
+              let data = image.pngData() else { return }
+        FileManager.default.createFile(
+            atPath: fileName,
+            contents: data,
+            attributes: nil
+        )
+    }
+
+    private func getImageFromCache(url: String) -> UIImage? {
+        guard let fileName = getFilePath(url: url),
+              let info = try? FileManager.default.attributesOfItem(
+                  atPath:
+                  fileName
+              ),
+              let modificationDate = info[FileAttributeKey.modificationDate] as?
+              Date
+        else { return nil }
+        let lifeTime = Date().timeIntervalSince(modificationDate)
+        guard
+            lifeTime <= cacheLifeTime,
+            let image = UIImage(contentsOfFile: fileName) else { return nil }
+        DispatchQueue.main.async {
+            self.images[url] = image
+        }
+        return image
+    }
+
+    private func loadPhoto(atIndexPath indexPath: IndexPath, byUrl url: String) {
+        AF.request(url).responseData(queue: DispatchQueue.global()) { [weak self] response in
+            guard
+                let data = response.data,
+                let image = UIImage(data: data) else { return }
+            DispatchQueue.main.async {
+                self?.images[url] = image
+            }
+            self?.saveImageToCache(url: url, image: image)
+            DispatchQueue.main.async {
+                self?.container.reloadRow(atIndexPath: indexPath)
+            }
+        }
```

Comment:
```suggestion
        AF.request(url).responseData(queue: DispatchQueue.global()) { [weak self] response in
            guard
                let data = response.data,
                let image = UIImage(data: data) else { return }
            DispatchQueue.main.async {
                self?.images[url] = image
            }
            self?.saveImageToCache(url: url, image: image)
            DispatchQueue.main.async {
                self?.container.reloadRow(atIndexPath: indexPath)
            }
        }
```
Возможно вынести в networkService